modules/bibcheck/lib/plugins/move_pattern_to_field.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+##
+## This file is part of Invenio.
+## Copyright (C) 2017 CERN.
+##
+## Invenio is free software; you can redistribute it and/or
+## modify it under the terms of the GNU General Public License as
+## published by the Free Software Foundation; either version 2 of the
+## License, or (at your option) any later version.
+##
+## Invenio is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Invenio; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""
+Bibcheck plugin to move values of fields matching some pattern to a new field.
+
+The new value is taken from the 'value' named group of 'pattern' (with the
+'(?P<value>regexp)' construct), if it matches the value of source_field and put
+into new_field. A subfield_filter can be specified to further restrict the
+source field and additional_subfields can optionally be set to a list of
+constant subfield (code, value) pairs to add.
+
+If there is a match, the original field is deleted. By default, the new field
+is only created if it does not result in a duplicate value. This behavior can
+be disabled by setting allow_duplicates to True.
+
+Example rules:
+
+    [move_KEKSCAN]
+
+    check = move_pattern_to_field
+    check.source_field = "8564_u"
+    check.new_field = "035__a"
+    check.pattern = "^(https?://)?www-lib.kek.jp/cgi-bin/img_index\\?(?P<value>(\\d{2})?\\d{7})$"
+    check.subfield_filter = ["y", "KEKSCAN"]
+    check.additional_subfields = [["9", "KEKSCAN"]]
+    check.allow_duplicates = false
+    filter_collection = HEP
+    filter_pattern = 8564_u:'kek.jp/cgi-bin' - 035__9:KEKSCAN
+
+"""
+
+
+import re
+
+from invenio.bibrecord import record_add_field, record_get_field_values
+
+def check_record(record, source_field, new_field, pattern, 
+                 subfield_filter=(None, None), additional_subfields=[], 
+                 allow_duplicates=False):
+
+    assert len(source_field) == 6
+    assert len(new_field) == 6
```

Comment:
I don't think it would be a problem, first as `record_add_field` is called before `delete_field`. The first appends a new field to the end of the `record[new_field[:3]]` list, so does not mess with the positions.